docs/src/diagnostics/users_diagnostics.md

```diff
@@ -0,0 +1,133 @@
+# Using ClimaLand Diagnostics when running a simulation
+
+When running a ClimaLand simulations, you have multiple options on how to write the outputs of that simulation. 
+You may want all variables, or just a selected few. 
+You may want instantaneous values, at the highest temporal and spatial resolution, or you may want to get averages at hourly or monthly time scale, and integrate in space 
+(for example soil moisture from 0 to 1 meter depth). 
+You may want to get more specific reductions, such as 10 days maximums, or compute a new variables that is a function of others. 
+You may want to get your outputs in memory in a Julia Dict, or write them in a NetCDF file. 
+
+This is where ClimaLand Diagnostics comes in for users. 
+
+In this documentation page, we first explain how to use default diagnostics and what are the defaults, and then explain how to define your own diagnostics for more advanced
+users.  
+
+# Default Diagnostics 
+
+Once you have defined your model and are ready to run a simulation, and after adding ClimaDiagnostics (using ClimaDiagnostics),
+ you can add default diagnostics to it by doing the following steps:
+
+1. define an output folder
+
+```
+mkdir("output")
+output_dir = "output/"
+```
+
+you may want to check if this directory exist and remove it if so, so that when you re-run the simulation, you start fresh:
+
+```
+if isdir("output")
+    rm("output", force = true, recursive = true)
+end
+
+mkdir("output")
+output_dir = "output/"
+```
+
+2. define a space
+
+Your diagnostics will be written in time and space. These may be defined in your model, but usually land model space is a sphere with no vertical dimension. 
+You may have variables varying with soil depth, and so you will need:
+
+```
+space = bucket_domain.space.subsurface
+```
+
+3. define your writter
+
+Your diagnostics will be written in a Julia Dict or a netcdf file, for example. This is up to you. For a netcdf file, you define your writter like this:
+
+```
+nc_writer = ClimaDiagnostics.Writers.NetCDFWriter(space, output_dir)
+```
+
+providing the space and output_dir defined in steps 1. and 2. 
+
+4. make your diagnostics on your model, using your writter, and define a callback
+
+Now that you defined your model and your writter, you can create a callback function to be called when solving your model. For example:
+
+```
+diags = ClimaLand.CLD.default_diagnostics(model, 1.0; output_writer = nc_writer)
+
+diagnostic_handler =
+    ClimaDiagnostics.DiagnosticsHandler(diags, Y, p, t0; dt = Δt)
+
+diag_cb = ClimaDiagnostics.DiagnosticsCallback(diagnostic_handler)
+
+sol = SciMLBase.solve(prob, ode_algo; dt = Δt, callback = diag_cb)
+```
+
+Your diagnostics have now been written in netcdf files in your output folder. 
+

```

Comment:
What if I want to compute some existing diagnostics but with different Writers? Or with different frequencies?